app/routes/join.tsx

```diff
@@ -66,7 +66,7 @@ export const action: ActionFunction = async ({ request }) => {
     request,
     userId: user.id,
     remember: false,
-    redirectTo: typeof redirectTo === "string" ? redirectTo : "/",
+    redirectTo,
```

Comment:
Maybe inline the call, so we're first doing redirects before "calculating" it?

```suggestion
    redirectTo: safeRedirect(formData.get("redirectTo"), "/"),
```